mat64/triangular.go

```diff
@@ -395,6 +396,76 @@ func (t *TriDense) MulTri(a, b Triangular) {
 	}
 }
 
+// Exp calculates the exponential of the matrix a, e^a, placing the result
+// in the receiver. If a has an odd number of rows, the matrix will be converted
+// to a Dense to perform the calculation.
+func (t *TriDense) Exp(a Triangular) {
+	n, kind := a.Triangle()
+	t.reuseAs(a.Triangle())
+
+	aU, _ := untransposeTri(a)
+	t.reuseAs(n, kind)
+	var restore func()
```

Comment:
Ah. This is because of the trickery you're pulling with NewTriDense (and I'm suggesting gets removed).

You're modifying the pointer to the receiver, but not the pointer to the input. If one has `m.Mul(m)`, the expectation would be that the data to `m` would change. Similarly, if one had
````
m := mat64.NewTriDense(blah)
a := m
m.Exp(a)
````
One would expect that the data for both a and m to change since they are pointers to the same data. This is the test you're failing, since you're changing the reference to `m` which breaks that `m` and `a` share the same data.